SYCL/ESIMD/tpm_pointer_v2.cpp

```diff
@@ -0,0 +1,206 @@
+//==---------------- pointer_tpm_v2.cpp  - DPC++ ESIMD on-device test
+//------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// TODO enable on Windows and Level Zero
+// REQUIRES: linux && gpu && opencl
+// RUN: %clangxx-esimd -fsycl %s -o %t.out
+// RUN: %HOST_RUN_PLACEHOLDER %t.out
+// RUN: %ESIMD_RUN_PLACEHOLDER %t.out
+
```

Comment:
all comments from the previous test apply here

Reply:
all tests seem to have a lot of overlap. Can they be merged into one using templates/parameters, so that code is reused as much as possible?